hydromt_wflow/wflow.py

```diff
@@ -239,25 +255,17 @@ def setup_basemaps(
         if geom is not None and geom.crs is None:
             raise ValueError("wflow region geometry has no CRS")
 
+        # Set the basins geometry
         ds_org = ds_org.raster.clip_geom(geom, align=res, buffer=10)
         ds_org.coords["mask"] = ds_org.raster.geometry_mask(geom)
         self.logger.debug("Adding basins vector to geoms.")
-        self.set_geoms(geom, name="basins")
 
-        # Check on resolution (degree vs meter) depending on ds_org res/crs
-        scale_ratio = int(np.round(res / ds_org.raster.res[0]))
-        if scale_ratio < 1:
-            raise ValueError(
-                f"The model resolution {res} should be \
-larger than the {hydrography_fn} resolution {ds_org.raster.res[0]}"
-            )
-        if ds_org.raster.crs.is_geographic:
-            if res > 1:  # 111 km
-                raise ValueError(
-                    f"The model resolution {res} should be smaller than 1 degree \
-(111km) for geographic coordinate systems. "
-                    "Make sure you provided res in degree rather than in meters."
-                )
+        # Set name based on scale_factor
+        basins_suffix = ""
+        if scale_ratio != 1:
+            basins_suffix = "_highres"
+        self.set_geoms(geom, name=f"basins{basins_suffix}")
```

Comment:
I think you can simplify and actually only set if scale_ratio != 1.
And right after that call self.basins to set it for all cases

Reply:
That seems fair!